### PR TITLE
[issue 3838] [schema] : Allow incompatible schemas to co-exist on a topic

### LINKED ISSUE #3838: Allow incompatible schemas to co-exist on a topic, allow set-schema-autoupdate-strategy = none
**Is your feature request related to a problem? Please describe.**
The set-schema-autoupdate-strategy policy currently allows disabling schema update (--disabled) and one of --compatibility(FULL, BACKWARD, FORWARD). There is no way to currently allow multiple schemas for a topic.

**Describe the solution you'd like**
 It would be useful to allow --compatibility so the schemas can be added without checking with the previous schemas. This allows for extra flexibility for instance having multiple objects on a topic, supports use cases like a simple event bus etc.

**Describe alternatives you've considered**
There is no workaround to achieve multiple distinct objects with their own schemas on one topic as of now.

**Additional context**
None

### PR DESCRIPTION
**Pull request by shiv4289**
## Problem:

Fixes #3838 : The set-schema-autoupdate-strategy policy currently allows disabling schema update (--disabled) and one of --compatibility(FULL, BACKWARD, FORWARD). There is no way to currently allow multiple schemas for a topic.

## Solution:
It would be useful to allow --compatibility so the schemas can be added without checking with the previous schemas. This allows for extra flexibility for instance having multiple objects on a topic, supports use cases like a simple event bus etc.

### Modifications

This change added tests and can be verified as follows:

  - *Added test cases under SchemaUpdateStrategyTest*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - The public API: (yes)
  - The schema: (yes)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (yes)
  - The admin cli options: (yes)
  - Anything that affects deployment: (no)

### Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (docs / JavaDocs added)

### CODE REVIEW

**Comment by shiv4289** [comment]
@sijie Hi, here is the patch as promised on slack. Please review this and let me know your comments.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/AlwaysSchemaValidator.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import org.apache.avro.SchemaValidationException;
+import org.apache.avro.SchemaValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An avro schema validator that always reports as compatible.
+ */
+class AlwaysSchemaValidator implements SchemaValidator {
+    private final static Logger log = LoggerFactory.getLogger(AlwaysSchemaValidator.class);
```
Comment (merlimat) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/AlwaysSchemaValidator.java]: nit: logger not used here

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/AlwaysSchemaValidator.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    static AlwaysSchemaValidator INSTANCE = new AlwaysSchemaValidator();
```
Comment (merlimat) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/AlwaysSchemaValidator.java]: nit: `static final ..`

Comment (shiv4289) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/AlwaysSchemaValidator.java]: Removed the logger, wasn't used. Forgot to remove after copy paste.. Great to spot the little one 👍

**Comment by merlimat** [comment]
@shiv4289 There's one header missing from the newly added file: 

```
2019-03-16\T\19:37:30.279 [WARNING] Missing header in: /home/jenkins/jenkins-slave/workspace/pulsar_precommit_java8/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/AlwaysSchemaValidator.java
```

You can use `mvn license:format` to automatically fix these.

**Comment by shiv4289** [comment]
> You can use `mvn license:format` to automatically fix these.

Thats magic, thank you. Done!

**Comment by jiazhai** [comment]
run integration tests

File: site2/website/versioned_docs/version-2.3.0/reference-pulsar-admin.md [hunk 3]
```diff
@@ -1296,7 +1296,7 @@ $ pulsar-admin namespaces set-schema-autoupdate-strategy tenant/namespace option
 Options
 |Flag|Description|Default|
 |----|---|---|
-|`-c`, `--compatibility`|Compatibility level required for new schemas created via a Producer. Possible values (Full, Backward, Forward).||
+|`-c`, `--compatibility`|Compatibility level required for new schemas created via a Producer. Possible values (Full, Backward, Forward, None).||
```
Comment (jiazhai) [inline site2/website/versioned_docs/version-2.3.0/reference-pulsar-admin.md]: remove this. since 2.3.0  has released and not include this type.

Comment (sijie) [reply site2/website/versioned_docs/version-2.3.0/reference-pulsar-admin.md]: 👍 Nice catch ! @jiazhai 

@shiv4289 can you remove this from 2.3.0 doc?

Comment (shiv4289) [reply site2/website/versioned_docs/version-2.3.0/reference-pulsar-admin.md]: Ack

**Comment by sijie** [comment]
run cpp tests

**Comment by sijie** [comment]
run cpp tests

**Comment by merlimat** [comment]
Merged in 2.3.1 at 
5dd2f47

**Pull request merged**